sound/soc/codecs/rt711.c

```diff
@@ -843,9 +843,16 @@ static const struct snd_soc_dapm_route rt711_audio_map[] = {
 static int rt711_set_bias_level(struct snd_soc_component *component,
 				enum snd_soc_bias_level level)
 {
+	struct rt711_priv *rt711 = snd_soc_component_get_drvdata(component);
 	struct snd_soc_dapm_context *dapm =
 		snd_soc_component_get_dapm(component);
 
+	if (!rt711->hw_init) {
+		dev_err(component->dev,
+			"%s called before hw_init completion\n", __func__);
+		return -EINVAL;
+	}
+
```

Comment:
@bardliao I am not sure...

1. the codec probe but nothing happens
2. the update_status callback is invoked and the initialization takes place

So when is the set_bias_level() called? if it's after 1) then we have a possible race condition.

Reply:
set_bias_level() will be called by dapm. So it will only be called after sound card is attached.

> 1. the codec probe but nothing happens

The probe here means rt711_sdw_probe, right?
rt711_probe will only success when hw_init is true which will be done by the update_status callback.